openpype/hosts/maya/api/lib_rendersettings.py

```diff
@@ -300,3 +306,79 @@ def _additional_attribs_setter(self, additional_attribs):
+        cmds.setAttr("defaultResolution.height", height)
+        cmds.setAttr("defaultRenderGlobals.animation", 1)
+
+        # frame range
+        start_frame = int(cmds.playbackOptions(query=True,
+                                               animationStartTime=True))
+        end_frame = int(cmds.playbackOptions(query=True,
+                                             animationEndTime=True))
+
+        dl_render_settings = cmds.ls(type="dlRenderSettings")
+        assert len(dl_render_settings) >= 1
+        for dl_render_setting in dl_render_settings:
+            cmds.setAttr(
+                "{}.startFrame".format(dl_render_setting), start_frame)
+            cmds.setAttr(
+                "{}.endFrame".format(dl_render_setting), end_frame)
+
+            # outputOptionsDefault
+            cmds.setAttr(
+                "{}.outputOptionsDefault".format(dl_render_setting), 2)
+
+        """
+        3delight doesn't use maya's render layers to indicate what to render,
+        instead, it uses dlRenderSettings as specified by the connection to
+        dlRenderGlobals1, of which there is only one. Since this is not the
+        "mayaonic" way, we must insert a pre-render MEL into maya's
+        "defaultRenderGlobals.preMel", which will help us work this out. We
+        wouldn't this this ... "hack" if deadline allowed us to insert commands
+        into its MayaBatch stuff, or, if it allowed setting up renderers in a
+        sensible way.
+        """
+
+        # We need the DOTALL since the regex will now match across newlines.
+        regex_match = re.compile(r"\/\*--\*3dl_v\d+\*--\*\/.*\/\*--\*\*--\*\/",
+                                 re.DOTALL)
+        PREMEL_TEMPLATE = """/*--*3dl_v4*--*/
+currentTime -e `getAttr ("defaultRenderGlobals.startFrame")`;
+string $lsCon[] = `listConnections renderLayerManager.renderLayerId`;
+for ($i = 0 ; $i<size($lsCon) ; $i++)
+{
+    string $cl = $lsCon[$i];
+    if (!startsWith($cl, "rs_"))
+        continue;
+    if (!endsWith($cl, "_RL"))
+        continue;
+    string $_3l = substring($cl, 4, size($cl)-3);
+    if (getAttr($cl+".renderable"))
+    {
+        string $rg = "dlRenderGlobals1.renderSettings";
+        DL_disconnectNode( $rg );
+        DL_connectNodeToMessagePlug( $_3l, $rg );
+        setAttr($_3l+".startFrame",
+            `getAttr ("defaultRenderGlobals.startFrame")`);
+        setAttr($_3l+".endFrame",
+            `getAttr ("defaultRenderGlobals.endFrame")`);
+        setAttr($_3l+".isRenderingSequence", 1);
+    }
+}
+/*--**--*/"""
+
+        preMel = cmds.getAttr("defaultRenderGlobals.preMel")
+        print("This is our current 'preMel':[{}]".format(preMel))
+        if preMel is None:
+            print("  - we need to insert our own preMel")
+            preMel = PREMEL_TEMPLATE
+        else:
+            print("  - we have preMel update it")
+            preMel = re.sub(regex_match, PREMEL_TEMPLATE, preMel)
```

Comment:
This is invalid - a premel script might already exist that MAY not contain the original premel script to update.
This should first do a `re.search` and if not found to APPEND the script.

Reply:
I guess this works by updating only its own part using regex

Reply:
But if there's no match than it doesn't add it? There's something invalid here. It's either expecting it to be there to get updated or for premel to be empty. It's not expecting the case the premel doesn't have the script yet but it does have some other value. Right?